source/Editor.js

```diff
@@ -1993,11 +1993,49 @@ var keyHandlers = {
             // If the commonAncestor was in a mathjax equation, delete the parent. Namely the mathjax equation.
             if (range.commonAncestorContainer && 
                 range.commonAncestorContainer.parentElement && 
-                checkInSVG.test(range.commonAncestorContainer.nodeName)
+                (checkInSVG.test(range.commonAncestorContainer.nodeName) || checkInSVG.test(range.commonAncestorContainer.parentElement.nodeName))
```

Comment:
More like

````javascript
var currentSearchNode = range.commonAncestorContainer;
var mathjaxParent = null;
while (currentSearchNode !== self._doc.body) {
    if (currentSearchNode.classList.contains('mathjax')) {
        mathjaxParent = currentSearchNode;
        break;
    }
    currentSearchNode = currentSearchNode.parent;
}

if (mathjaxParent) {
    event.preventDefault();
    detach(mathjaxParent);
}
````

i.e. it looks like you want to prevent default if the common ancestor is in a span.mathjax.